apps/ejabberd/src/foreign_event/mod_foreign.erl

```diff
@@ -73,6 +79,20 @@ get_disco_items(empty, From, To, Node, Lang) ->
 get_disco_items(Acc, _From, _To, _Node, _Lang) ->
     Acc.
 
+-spec iq_handler(From :: jid(), To :: jid(), iq()) -> iq() | ignore.
+iq_handler(_From, _To, #iq{type = get} = IQ) ->
+    IQ#iq{type = error, sub_el = ?ERR_NOT_ALLOWED};
+iq_handler(_From, _To, #iq{type = set, sub_el = ForeignEvent} = IQ) ->
+    case parse_foreign_event(ForeignEvent) of
+        {ok, Type, Request, _Publish} ->
+            maybe_dispatch_request(Type, Request, IQ);
+        error ->
```

Comment:
{error, missing_request_element} | {error, missing_type_field} is more preferable.